crates/pallet-wrapped-token/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+    fn name() -> &'static str;
+
+    /// Returns the symbol of the token.
+    fn symbol() -> &'static str;
+
+    /// Returns the decimals places of the token.
+    fn decimals() -> u8;
+}
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+/// Utility alias for easy access to the [`Config::AccountId`].
+type AccountIdOf<T, I> = <T as Config<I>>::AccountId;
+
+/// Utility alias for easy access to the [`Currency::Balance`] of the [`Config::Currency`] type.
+type BalanceOf<T, I> = <<T as Config<I>>::Currency as Currency<AccountIdOf<T, I>>>::Balance;
+
+// We have to temporarily allow some clippy lints. Later on we'll send patches to substrate to
+// fix them at their end.
+#[allow(clippy::missing_docs_in_private_items)]
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{pallet_prelude::*, sp_runtime::traits::MaybeDisplay, sp_std::fmt::Debug};
+
+    use super::*;
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T, I = ()>(_);
+
+    /// Configuration trait of this pallet.
+    #[pallet::config]
+    pub trait Config<I: 'static = ()>: frame_system::Config {
+        /// The user account identifier type.
+        type AccountId: Parameter
+            + Member
+            + MaybeSerializeDeserialize
+            + Debug
+            + MaybeDisplay
+            + Ord
+            + MaxEncodedLen;
+
+        /// The currency to be exposed as ERC20 token.
+        type Currency: Currency<AccountIdOf<Self, I>>;
+
+        /// Interface into ERC20 metadata implementation.
+        type Metadata: Metadata;
+    }
+
+    /// ERC20-style approvals data.
+    /// (Owner => Allowed => Amount).
+    #[pallet::storage]
+    #[pallet::getter(fn approvals)]
+    pub type Approvals<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        AccountIdOf<T, I>,
+        Blake2_128Concat,
+        AccountIdOf<T, I>,
+        BalanceOf<T, I>,
+    >;
+
+    /// Possible errors.
+    #[pallet::error]
+    pub enum Error<T, I = ()> {
+        /// Spender not allowed to transfer tokens on behalf of owner.
+        SpenderNotAllowed,
+        /// Spender can't transfer tokens more than allowed.
+        SpendMoreThanAllowed,
+    }
+}
+
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+    /// Returns the amount of tokens in existence.
+    pub fn total_supply() -> BalanceOf<T, I> {
+        T::Currency::total_issuance()
+    }
+
+    /// Returns the amount of tokens owned by provided account.
+    pub fn balance_of(owner: &AccountIdOf<T, I>) -> BalanceOf<T, I> {
+        T::Currency::total_balance(owner)
+    }
+
+    /// Returns the remaining number of tokens that spender will be allowed to spend on behalf of
+    /// owner. This is zero by default.
+    pub fn allowance(owner: &AccountIdOf<T, I>, spender: &AccountIdOf<T, I>) -> BalanceOf<T, I> {
```

Comment:
I think there are aliases that can do just this, can't they? They might be deprecated in the substrate `master` though, so not sure if we want to use them... But just to note this. Maybe double-check if they are indeed deprecated, and then use them if they are not?

Reply:
https://github.com/humanode-network/humanode/pull/746/commits/654dc6523c41bb6598fa23e34b3ca80dcec31b42

Reply:
This is not what I meant...

Reply:
Reverted - https://github.com/humanode-network/humanode/pull/746/commits/66f9c506b78259c18924508587679968da313511.